src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1193,7 +1194,7 @@ public string DnsSafeHost
                 if (HostType == Flags.IPv6HostType)
                 {
                     ret = ret.Substring(1, ret.Length - 2);
-                    if ((object)_info.ScopeId != null)
+                    if ((object)_info.ScopeId != null && (_flags & Flags.IPv6LinkLocalAddress) == 0)
```

Comment:
Nit: We can add a comment here to explain the purpose for the new flag. i,e,. to avoid duplicated `ScopeId`.

Reply:
Added